File: 프로그래머스/3/42898. 등굣길/등굣길.java
import java.util.*;

class Solution {
    public int solution(int m, int n, int[][] puddles) {
        
        int map[][] = new int[n+1][m+1];
        boolean isPuddle[][] = new boolean[n+1][m+1];
        
        for(int i=0; i<puddles.length; i++){
            isPuddle[puddles[i][1]][puddles[i][0]] = true;
        }
        
        map[1][1] = 1;
        
        for(int i=1; i<=n; i++){
            for(int j=1; j<=m; j++){
                if(isPuddle[i][j]) continue;
                map[i][j] += map[i-1][j]+map[i][j-1];
                map[i][j] %= 1000000007;
            }
        }
       
        return map[n][m];
            
    }
}

File: 프로그래머스/3/42898. 등굣길/등굣길.js
function solution(m, n, puddles) {
    
    const map = Array.from({length : n+1}, () => Array(m+1).fill(0));
    
    puddles.forEach((x) => {
        map[x[1]][x[0]] = -1;
    })
    
    map[1][1] = 1;
    
    for(let i=1; i<= n; i++){
        for(let j=1; j<=m; j++){
            if(map[i][j] == -1) continue
            if(map[i-1][j] > 0) map[i][j] += map[i-1][j]%1000000007
            if(map[i][j-1] > 0) map[i][j] += map[i][j-1]%1000000007
            map[i][j] %= 1000000007
        }
    }
    
    
    return map[n][m];
}